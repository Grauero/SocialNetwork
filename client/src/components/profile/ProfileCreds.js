import React from 'react';
import propTypes from 'prop-types';
import Moment from 'react-moment';

const ProfileCreds = ({ experience, education }) => {
  const experienceItems = experience.map(exp => (
    <li key={exp._id} className="list-group-item">
      <h4>{exp.company}</h4>
      <p>
        <Moment format="YYYY/MM/DD">{exp.from}</Moment> -
        {exp.to === null ? (
          'Now'
        ) : (
          <Moment format="YYYY/MM/DD" data-to>
            {exp.to}
          </Moment>
        )}
      </p>
      <p>
        <strong>Position: </strong>
        {exp.title}
      </p>
      <p>
        {exp.location === '' ? null : (
          <span data-location>
            <strong>Location: </strong>
            {exp.location}
          </span>
        )}
      </p>
      <p>
        {exp.description === '' ? null : (
          <span data-description>
            <strong>Description: </strong>
            {exp.description}
          </span>
        )}
      </p>
    </li>
  ));
  const educationItems = education.map(edu => (
    <li key={edu._id} className="list-group-item">
      <h4>{edu.school}</h4>
      <p>
        <Moment format="YYYY/MM/DD">{edu.from}</Moment> -
        {edu.to === null ? (
          'Now'
        ) : (
          <Moment format="YYYY/MM/DD">{edu.to}</Moment>
        )}
      </p>
      <p>
        <strong>Degree: </strong>
        {edu.degree}
      </p>
      <p>
        <strong>Field Of Study: </strong>
        {edu.fieldOfStudy}
      </p>
      <p>
        {edu.description === '' ? null : (
          <span data-description>
            <strong>Description: </strong>
            {edu.description}
          </span>
        )}
      </p>
    </li>
  ));

  return (
    <div className="row">
      <div className="col-md-6">
        <h3 className="text-center text-info">Experience</h3>
        {experienceItems.length > 0 ? (
          <ul className="list-group" data-experience>
            {experienceItems}
          </ul>
        ) : (
          <p className="text-center">No experience provided</p>
        )}
      </div>

      <div className="col-md-6">
        <h3 className="text-center text-info">Education</h3>
        {educationItems.length > 0 ? (
          <ul className="list-group" data-education>
            {educationItems}
          </ul>
        ) : (
          <p className="text-center">No education provided</p>
        )}
      </div>
    </div>
  );
};

ProfileCreds.propTypes = {
  experience: propTypes.instanceOf(Array).isRequired,
  education: propTypes.instanceOf(Array).isRequired
};

export default ProfileCreds;
