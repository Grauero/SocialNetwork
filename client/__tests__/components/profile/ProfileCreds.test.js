import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ProfileCreds from '../../../src/components/profile/ProfileCreds';

const props = {
  experience: [
    {
      _id: 'id',
      company: 'company',
      from: 'from',
      to: 'to',
      title: 'title',
      location: 'location',
      description: 'description'
    }
  ],
  education: [
    {
      _id: 'id',
      school: 'school',
      from: 'from',
      to: 'to',
      degree: 'degree',
      fieldOfStudy: 'field',
      description: 'description'
    }
  ]
};
let component;

beforeEach(() => (component = shallow(<ProfileCreds {...props} />)));

afterEach(() => component.unmount());

function generateTestForProps(category, param) {
  it(`NOT renders ${param} if its not defined`, () => {
    const testProps = Object.assign({}, props, {
      [category]: [{ ...props[category][0], [param]: '' }]
    });

    component = shallow(<ProfileCreds {...testProps} />);
    const ul = component.find(`ul[data-${param}]`);
    const item = ul.find(`span[data-${param}]`);

    expect(item.debug()).toBe('');
  });
}

generateTestForProps('experience', 'location');
generateTestForProps('experience', 'description');
generateTestForProps('education', 'description');

it('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});

it('NOT renders any experience if experience is empty', () => {
  const testProps = Object.assign({}, props, { experience: [] });

  component = shallow(<ProfileCreds {...testProps} />);
  const ul = component.find('ul[data-experience]');

  expect(ul.debug()).toBe('');
});

it('NOT renders any education if education is empty', () => {
  const testProps = Object.assign({}, props, { education: [] });

  component = shallow(<ProfileCreds {...testProps} />);
  const ul = component.find('ul[data-education]');

  expect(ul.debug()).toBe('');
});
