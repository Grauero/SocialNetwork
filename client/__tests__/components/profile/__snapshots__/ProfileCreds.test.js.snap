// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6"
  >
    <h3
      className="text-center text-info"
    >
      Experience
    </h3>
    <ul
      className="list-group"
      data-experience={true}
    >
      <li
        className="list-group-item"
        key="id"
      >
        <h4>
          company
        </h4>
        <p>
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="YYYY/MM/DD"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            from
          </t>
           -
          <t
            ago={false}
            calendar={false}
            data-to={true}
            decimal={false}
            element="time"
            filter={[Function]}
            format="YYYY/MM/DD"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            to
          </t>
        </p>
        <p>
          <strong>
            Position: 
          </strong>
          title
        </p>
        <p>
          <span
            data-location={true}
          >
            <strong>
              Location: 
            </strong>
            location
          </span>
        </p>
        <p>
          <span
            data-description={true}
          >
            <strong>
              Description: 
            </strong>
            description
          </span>
        </p>
      </li>
    </ul>
  </div>
  <div
    className="col-md-6"
  >
    <h3
      className="text-center text-info"
    >
      Education
    </h3>
    <ul
      className="list-group"
      data-education={true}
    >
      <li
        className="list-group-item"
        key="id"
      >
        <h4>
          school
        </h4>
        <p>
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="YYYY/MM/DD"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            from
          </t>
           -
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="YYYY/MM/DD"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            to
          </t>
        </p>
        <p>
          <strong>
            Degree: 
          </strong>
          degree
        </p>
        <p>
          <strong>
            Field Of Study: 
          </strong>
          field
        </p>
        <p>
          <span
            data-description={true}
          >
            <strong>
              Description: 
            </strong>
            description
          </span>
        </p>
      </li>
    </ul>
  </div>
</div>
`;
